.github/workflows/stardis-tests.yml

```diff
@@ -0,0 +1,79 @@
+name: Stardis Tests
+
+on:
+  push:
+    branches:
+    - '*'
+
+  pull_request:
+    branches:
+    - '*'
+    
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - labeled
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash -l {0}
+ 
+env:
+  CACHE_NUMBER: 0  # increase to reset cache manually
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        label: [osx-64, linux-64]
+        include:
+          - os: ubuntu-latest
+            label: linux-64
+            prefix: /usr/share/miniconda3/envs/stardis
+
+          - os: macos-latest
+            label: osx-64
+            prefix: /Users/runner/miniconda3/envs/stardis
+
+    name: ${{ matrix.label }}
+    runs-on: ${{ matrix.os }}
+    steps:     
+      - uses: actions/checkout@v4
+        with:
+          repository: tardis-sn/stardis
+
+      - name: Setup Mambaforge
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          activate-environment: stardis
+          use-mamba: true
+  
+      - name: Cache lockfile
+        uses: actions/cache@v2
+        with:
+          path: ${{ matrix.prefix }}
+          key: conda-${{ matrix.label }}-${{ hashFiles(format('conda-{0}.lock', matrix.label)) }}-${{ env.CACHE_NUMBER }}
+        id: cache-conda
+  
+      - name: Update Conda Environment
+        id: update-env
+        run: |
+          mamba update -n stardis --file conda-${{ matrix.label }}.lock
+        if: steps.cache-conda.outputs.cache-hit != 'true'
+        
+      - name: Install TARDIS
+        id: install-tardis
+        run: |
+          pip install git+https://github.com/tardis-sn/tardis.git@${{ github.sha }}
```

Comment:
does this work when triggered from master, have you tried other test cases?

Reply:
Workflow Dispatch:
https://github.com/KasukabeDefenceForce/tardis/actions/runs/10143033368/job/28043570915

Pull Request:
https://github.com/KasukabeDefenceForce/tardis/pull/34